cpp/src/groupby/hash/groupby.cu

```diff
@@ -631,6 +618,16 @@ std::unique_ptr<table> groupby_null_templated(table_view const& keys,
 
 }  // namespace
 
+/**
+ * @brief Indicates whether the specified aggregation operation can be computed
+ * with a hash-based implementation.
+ *
+ * @param t The aggregation operation to verify
+ * @return true `t` is valid for a hash based groupby
+ * @return false `t` is invalid for a hash based groupby
+ */
+bool is_hash_aggregation(aggregation::Kind t) { return array_contains(hash_aggregations, t); }
```

Comment:
Separating the definition makes this no longer `constexpr`. If you want it available in the header, move it's definition there as well and keep it `constexpr`.